airflow/providers/cncf/kubernetes/hooks/kubernetes.py

```diff
@@ -347,6 +350,10 @@ def _get_namespace(self) -> str | None:
             return self._get_field("namespace")
         return None
 
+    def get_xcom_sidecar_container_image(self):
+        """Returns the xcom sidecar image that defined in the connection"""
+        return self._get_field("xcom_sidecar_container_image") or None
```

Comment:
```suggestion
        return self._get_field("xcom_sidecar_container_image")
```
this is handled by get_field